src/include/daos/common.h

```diff
@@ -104,12 +104,17 @@ char *DP_UUID(const void *uuid);
 #define DP_CONT(puuid, cuuid)	DP_UUID(puuid), DP_UUID(cuuid)
 #define DF_CONTF		DF_UUIDF"/"DF_UUIDF
 
+#ifndef DAOS_BUILD_RELEASE
 char *daos_key2str(daos_key_t *key);
 
 #define DF_KEY			"[%d] %.*s"
 #define DP_KEY(key)		(int)(key)->iov_len,	\
 		                (int)(key)->iov_len,	\
 		                daos_key2str(key)
+#else
```

Comment:
For the release builds then the daos_key2str() symbol will still be compiled as the compiler has no way of knowing it's not used.  If this was for the debug build I don't think it would matter, but we shouldn't have symbols we know aren't being used in the release build so can you remove the function as well.  It would act as a double-check it wasn't being called.

Reply:
I looked through the symbol table with nm and binutils when I compiled the release build and daos_key2str is not in the release binary at all. It is only in when the build type is debug and dev.

Reply:
I ran this in install/lib64 which is where the daos library and common library are on a scons with BUILD_TYPE=release.

for f in *.so; do nm $f | grep daos_key2str; done

It came up with 0 results. When the same line is run against dev and debug releases it shows up with 2 entries an Undefined symbol in libdaos and a text segment entry in libdaos_common.

Reply:
That's pretty cool, I wonder if it's because we use default visibility hidden?  That would mean the linker didn't export it, and knew it wasn't called across files so could remote it.

If that's actually the case then I wonder if we should try a test with and without visibility-hidden and see what other functions we have that aren't used.